src/core/skills/game_mode/pve/pve_classic_tianji.ts

```diff
@@ -0,0 +1,72 @@
+import { CardMoveReason, EventPacker, GameEventIdentifiers, ServerEventFinder } from 'core/event/event';
+import { DamageType } from 'core/game/game_props';
+import { AllStage, PhaseStageChangeStage, PlayerPhaseStages } from 'core/game/stage_processor';
+import { Player } from 'core/player/player';
+import { PlayerCardsArea } from 'core/player/player_props';
+import { Room } from 'core/room/room';
+import { TriggerSkill } from 'core/skills/skill';
+import { CompulsorySkill } from 'core/skills/skill_wrappers';
+import { TranslationPack } from 'core/translations/translation_json_tool';
+
+@CompulsorySkill({ name: 'pve_classic_tianji', description: 'pve_classic_tianji_description' })
+export class PveClassicTianJi extends TriggerSkill {
+  isTriggerable(event: ServerEventFinder<GameEventIdentifiers.PhaseStageChangeEvent>, stage?: AllStage) {
+    return stage === PhaseStageChangeStage.StageChanged;
+  }
+
+  canUse(room: Room, owner: Player, content: ServerEventFinder<GameEventIdentifiers.PhaseStageChangeEvent>) {
+    const noDamage =
+      room.Analytics.getRecordEvents<GameEventIdentifiers.DamageEvent>(
```

Comment:
as `canUse` will be called pretty frequently, searching in event analytics is a low-efficiency action, so I'd recommend to check the other conditions at first and then checking `noDamage`, it gonna reduce the calling of analytics.